apache/airflow
# Auto tail file logs in Web UI

This PR implements a log tail feature on file_task_handler

As described in #15783 it would be a great feature to have log tail available in Airflow UI so that if you watch something running you don't need to hit the F5 key constantly to follow the running task.

Actually I saw the same demand and with this PR want to contribute this feature upstream.

I wanted to implement this from scratch when I realized that an auto-tail is already implemented in the UI, just the back-end needs to feed data in a way that tail is active. So this PR is a minimal-invasive implementation on the `file_task_handler.py` alongside with small adjustments to inheriting provider implementation.

closes: #15783
related: #15783

How to test:
* Start a long running DAG based on an airflow instance from this branch and see that the running task instance which produces logs will add content to the view.

Example DAG I used for testing, running 200 seconds and adding log lines every second:
```
import pendulum

from airflow import DAG
from airflow.operators.bash import BashOperator

with DAG(
    dag_id='a_log_running_dag',
    schedule='0 * * * *',
    start_date=pendulum.datetime(2021, 1, 1, tz="UTC"),
    catchup=False,
) as dag:
    BashOperator(
        task_id='runme',
        bash_command='for i in {1..200}; do echo $i; sleep 1; done',
    )
```
